pkg/security/controller/namespace_scc_allocation_controller.go

```diff
@@ -194,40 +189,40 @@ func (c *NamespaceSCCAllocationController) allocate(ctx context.Context, syncCtx
 		return fmt.Errorf("%d not in range", bitIndex)
 	}
 
-	// Now modify the namespace
-	nsCopy := ns.DeepCopy()
-	if nsCopy.Annotations == nil {
-		nsCopy.Annotations = make(map[string]string)
+	// Update the namespace using server-side apply.
+	patch := &applycorev1.NamespaceApplyConfiguration{
+		TypeMetaApplyConfiguration: applymetav1.TypeMetaApplyConfiguration{
+			Kind:       ptr.To("Namespace"),
+			APIVersion: ptr.To("v1"),
+		},
+		ObjectMetaApplyConfiguration: &applymetav1.ObjectMetaApplyConfiguration{
+			Name:            ptr.To(ns.Name),
+			ResourceVersion: ptr.To(ns.ResourceVersion),
+			Annotations: map[string]string{
+				securityv1.UIDRangeAnnotation:           block.String(),
+				securityv1.SupplementalGroupsAnnotation: block.String(),
+			},
+		},
 	}
-	nsCopy.Annotations[securityv1.UIDRangeAnnotation] = block.String()
-	nsCopy.Annotations[securityv1.SupplementalGroupsAnnotation] = block.String()
-	if _, ok := nsCopy.Annotations[securityv1.MCSAnnotation]; !ok {
+	// We need to make sure all annotations are always set for server-side apply, otherwise some would get removed.
+	// The following code actually makes sure the annotation is always set once it is set, which is ok regarding SSA.
+	if v, ok := ns.Annotations[securityv1.MCSAnnotation]; ok {
+		patch.Annotations[securityv1.MCSAnnotation] = v
```

Comment:
`patch.ObjectMetaApplyConfiguration.Annotations` to avoid any "hmm, is it really embedded?" question.